src/libproc_macro/lib.rs

```diff
@@ -812,7 +812,7 @@ impl Ident {
     }
 
     /// Same as `Ident::new`, but creates a raw identifier (`r#ident`).
```

Comment:
One more line:
```rust
In addition to general identifier validity requirements, raw identifiers cannot be created for keywords usable in path segments (`self`, `super` and others).
```

Reply:
That leaves me somewhat confused; what does "general identifier validity requirements" entail here? Surely `Ident::new_raw` has fewer restrictions than `Ident::new_raw` but your wording suggests it has more -- in particular, when will `Ident::new_raw` panic?

Reply:
"general identifier validity requirements" -> "general identifier validity requirements inherited from `Ident::new`".
We can point to the reference or something that describes what is an identifier.

>Surely Ident::new_raw has fewer restrictions than Ident::new_raw but your wording suggests it has more

It does have more restrictions.

>in particular, when will Ident::new_raw panic?

"raw identifiers cannot be created for" -> "this function will panic on attempts to create"

Reply:
> It does have more restrictions.

Ah! `$ident` accepts keywords as well whereas it does not accept `r#self` -- that's pretty counter-intuitive. I think amending the documentation of `fn new` would be good as well to note the fact that keywords are allowed.